macos/Sources/App/macOS/AppDelegate.swift

```diff
@@ -485,10 +485,16 @@ class AppDelegate: NSObject,
         }
 
         // Sync our auto-update settings
-        updaterController.updater.automaticallyChecksForUpdates =
-            config.autoUpdate == .check || config.autoUpdate == .download
-        updaterController.updater.automaticallyDownloadsUpdates =
-            config.autoUpdate == .download
+        // Local (source) builds always disable the updater
+        if let commit = Bundle.main.infoDictionary?["GhosttyCommit"] as? String, !commit.isEmpty {
```

Comment:
We could also wrap this in a more generic "isLocalBuild" property under e.g. `Ghostty.Info` if we think it would be useful elsewhere or improve the clarity of this check.

Reply:
I'm not sure this is right. Because tip builds should have a commit too, no? I'd prefer to use something a bit more robust. I'm not totally sure what that is yet (there are multiple options...).

The one I'm leaning towards the most is inserting a key into the plist file for releases only (`release-tip` `release-tag` jobs) and only enabling sparkle at all if its present.

Reply:
> Because tip builds should have a commit too, no?

Yes, `GhosttyCommit` is set by all of the release workflows: pr, tag, tip. It is _not_ set as part of the local build process.

> The one I'm leaning towards the most is inserting a key into the plist file for releases only (release-tip release-tag jobs) and only enabling sparkle at all if its present.

I think that'll work. It's most similar to the third alternative I listed above. Do you have a preference for (ab)using [Sparkle's `SUEnableAutomaticChecks`](https://sparkle-project.org/documentation/customization/) for this or introducing our own `GhosttyAutoUpdate` key?

Reply:
Oh my mistake I misread the boolean logic.

And yes! `SUEnableAutomaticChecks` is what I was thinking of! I think for debug and release local schemes we should just set this to NO and for other schemes we should leave it unset (to allow users to override it).

We can do this in the `Ghostty-Info.plist` in Xcode.

Reply:
We unfortunately can't vary an Info.plist value on a per-scheme basis without taking one of the more elaborate approaches I listed above.

I think the compromise might be to default it to NO (which would apply to all build schemes) but toggle it back on in the release workflows where we want it.

Reply:
You can also set `INFOPLIST_PREPROCESS` to run the C preprocessor on `Info.plist`. Paired with `INFOPLIST_OTHER_PREPROCESSOR_FLAGS` conditionalized on the config, you could try something like this:

```
INFOPLIST_PREPROCESS = YES
INFOPLIST_OTHER_PREPROCESSOR_FLAGS[config=ReleaseLocal] = -DLOCAL=1
INFOPLIST_OTHER_PREPROCESSOR_FLAGS[config=Debug] = -DLOCAL=1
```

And then add this snippet to the `Info.plist`:

```
	<key>SUEnableAutomaticChecks</key>
#if LOCAL
	</false>
#else
	</true>
#endif
```

Reply:
Thanks for the suggestion, @mattwidmann! I used a `INFOPLIST_PREPROCESSOR_DEFINITIONS`-based approach in my latest revision.